tutorial overlay and middle button

yuemily24 opened the pull request:
# Description

- new tutorial overlay when the user finishes onboarding
- middle tab bar, when shown with a calendar icon, will _always_ default to Calendar tab now
<img width="363" alt="Screen Shot 2022-04-26 at 8 26 38 PM" src="https://user-images.githubusercontent.com/76456096/165414295-38371718-174f-49bf-b70b-8fb5ad3e3310.png">

Fixes # (issue/ticket number)

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (if required)
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

yuemily24 committed:
middle button always leads back to calendar tab

yuemily24 committed:
Merge branch 'develop' into emilys-fixes

yuemily24 committed:
new tutorial overlay

faye-tan commented on tpp-app/src/home/pages/CalendarScreen.js:
```diff
@@ -123,30 +125,37 @@ export default function CalendarScreen ({ route, navigation }) {
             }
 
+            // show tutorial overlay if coming from Confirmation screen, else ignore
+            AsyncStorage.getItem('showTutorial')
```
Can you turn this into a function to put in a /service file? It would be easier because in the .catch() of all service files should have the error popup when error is caught

yuemily24 replied:
Done!

faye-tan commented on tpp-app/src/home/pages/TutorialOverlay.js:
```diff
@@ -0,0 +1,89 @@
+export default function TutorialOverlay({ navigation }) {
+  const tabBarHeight = useBottomTabBarHeight();
+  // set 'showTutorial' key to false in backend
+  AsyncStorage.setItem('showTutorial', 'false')
```
Same here, and suggest a constant for 'showTutorial' :3

faye-tan commented on tpp-app/src/onboarding/Confirmation.js:
```diff
@@ -84,9 +87,13 @@ export default function Confirmation ({ route, navigation }) {
         title="Let's go!" color="#5A9F93" bottom="-8%"
-        onPress={() => navigation.navigate(STACK_SCREENS.MAIN_PAGE)}
+        onPress={() => {
+          AsyncStorage.setItem('showTutorial', 'true')
```
here as well!

yuemily24 committed:
TutorialService with backend helper functions

yuemily24 committed:
imported missing functions in Settings.js

yuemily24 committed:
Merge branch 'develop' into emilys-fixes

yuemily24 commented:
Added new `TutorialService.js` file where I moved all the backend functions related to the tutorial overlay and setting the true/false key for it. Also added missing imports throwing errors in `Settings.js`.

yuemily24 committed:
Merge branch 'emilys-fixes' of https://github.com/uoftblueprint/the-period-purse into emilys-fixes

yuemily24 committed:
fix PeriodLength textinput icon alignment

yuemily24 committed:
Merge branch 'develop' into emilys-fixes

faye-tan committed:
Merge branch 'develop' into emilys-fixes
